lib/src/runner/node/platform.dart

```diff
@@ -96,29 +96,35 @@ class NodePlatform extends PlatformPlugin
       TestPlatform platform, SuiteConfiguration suiteConfig) async {
     var server = await MultiServerSocket.loopback(0);
 
-    var pair = await _spawnProcess(path, platform, suiteConfig, server.port);
-    var process = pair.first;
-
-    // Forward Node's standard IO to the print handler so it's associated with
-    // the load test.
-    //
-    // TODO(nweiz): Associate this with the current test being run, if any.
-    process.stdout.transform(lineSplitter).listen(print);
-    process.stderr.transform(lineSplitter).listen(print);
-
-    var socket = await server.first;
-    // TODO(nweiz): Remove the DelegatingStreamSink wrapper when sdk#31504 is
-    // fixed.
-    var channel = new StreamChannel(socket, new DelegatingStreamSink(socket))
-        .transform(new StreamChannelTransformer.fromCodec(UTF8))
-        .transform(chunksToLines)
-        .transform(jsonDocument)
-        .transformStream(new StreamTransformer.fromHandlers(handleDone: (sink) {
-      if (process != null) process.kill();
-      sink.close();
-    }));
-
-    return new Pair(channel, pair.last);
+    try {
+      var pair = await _spawnProcess(path, platform, suiteConfig, server.port);
+      var process = pair.first;
+
+      // Forward Node's standard IO to the print handler so it's associated with
+      // the load test.
+      //
+      // TODO(nweiz): Associate this with the current test being run, if any.
+      process.stdout.transform(lineSplitter).listen(print);
+      process.stderr.transform(lineSplitter).listen(print);
+
+      var socket = await server.first;
+      // TODO(nweiz): Remove the DelegatingStreamSink wrapper when sdk#31504 is
+      // fixed.
+      var channel = new StreamChannel(socket, new DelegatingStreamSink(socket))
+          .transform(new StreamChannelTransformer.fromCodec(UTF8))
+          .transform(chunksToLines)
+          .transform(jsonDocument)
+          .transformStream(
+              new StreamTransformer.fromHandlers(handleDone: (sink) {
+        if (process != null) process.kill();
+        sink.close();
+      }));
+
+      return new Pair(channel, pair.last);
+    } catch (_) {
+      server.close().catchError((_) {});
```

Comment:
Should we add additional messaging here notifying that the error is coming from the node compilation or will that be obvious from the stderr and current error?

Reply:
I don't think it's actually possible for this to throw, and if it does we don't actually care about the error. I just have `catchError()` here to be absolutely sure that it doesn't confuse the user.